libvcx/src/connection.rs

```diff
@@ -854,25 +856,20 @@ pub fn send_generic_message(connection_handle: u32, msg: &str, msg_options: &str
     })
 }
 
-pub fn update_state_with_message(handle: u32, message: Message) -> VcxResult<u32> {
+pub fn update_state_with_message(handle: u32, message: A2AMessage) -> VcxResult<u32> {
```

Comment:
Nice refactor, narrowing down necessary information from `Message` which contains agency information to only `A2AMessage`, hence only protocol messages, forgetting existence of Agency.